src/f4enix/input/MCNPinput.py

```diff
@@ -1339,12 +1330,85 @@ def hash_multiple_cells(self, hash_dict: dict[int, int]) -> None:
                 cell = self.cells[str(cell_num)]
                 self.hash_cell(cell, hash_id, inplace=True)
 
+    @staticmethod
+    def cell_union_function(
+        cell_list: list[parser.Card],
+        new_cell_num: int = None,
+    ) -> parser.Card:
+        """Given a list of cells, it creates a new cell that is the union of
+        the cells in the list.
+
+        Parameters
+        ----------
+        cell_list : list[parser.Card]
+            list of cells to be united
+        new_cell_num : int, optional
+            new number of the union cell, by default None
+        """
+
+        if new_cell_num is None:
+            cel = cell_list[0]
+            new_cell_num = cel.values[0][0]
+
+        new_cell = cell_list[0]
+
+        for cell in cell_list[1:]:
+            new_cell = Input._add_symbol_to_cell_input(
+                new_cell,
+                " : (" + cell.get_geom().replace("\n", " ") + ")",
+                inplace=False,
+                parentheses=True,
+                new_cell_num=new_cell_num,
+            )
+            cell_lines = new_cell.card(wrap=True).splitlines(keepends=True)
+            new_cell = parser.Card(cell_lines, 3, new_cell.pos)
+            new_cell.get_values()
+
+        return new_cell
+
+    def cells_union(
+        self,
+        cell_list: list[parser.Card],
+        remove_cells: bool = True,
+        add_union_cell: bool = True,
+        new_cell_num: int = None,
+    ) -> None:
+        """Given a list of cells, it creates a new cell that is the union of
+        the cells in the list. It also checks if the original cells must be removed or not
+        and if the new one has to be added to the input. If a renumbering was done,
+        the input must be re-read.
+
+        Parameters
+        ----------
+        cell_list : list[parser.Card]
+            list of cells to be united
+        remove_cells : bool, optional
+            Flag to check if the united cells have to be removed, by default True
+        add_union_cell : bool, optional
+            flag to check if the resultant cell has to be added to the input, by default True
+        new_cell_num : int, optional
+            new number of the union cell, by default None
+        """
+
+        if new_cell_num is None:
+            cel = cell_list[0]
+            new_cell_num = cel.values[0][0]
+
```

Comment:
If ''remove_cells'' is false, then the the starting cell on which to build the union should be a copy in order not to modify it inplace. Add a test for that also